Mathlib/Data/Fin/Basic.lean

```diff
@@ -438,84 +438,83 @@ theorem eq_succ_of_ne_zero {n : ℕ} {i : Fin (n + 1)} (hi : i ≠ 0) : ∃ j :
   (eq_zero_or_eq_succ i).resolve_left hi
 #align fin.eq_succ_of_ne_zero Fin.eq_succ_of_ne_zero
 
+theorem rev_involutive : Involutive (rev : Fin n → Fin n) := fun i =>
+  ext <| by
+    dsimp only [rev]
+    rw [← tsub_tsub, tsub_tsub_cancel_of_le (Nat.add_one_le_iff.2 i.is_lt),
+      add_tsub_cancel_right]
+#align fin.rev_involutive Fin.rev_involutive
+
 /-- `Fin.rev` as an `Equiv.Perm`, the antitone involution `Fin n → Fin n` given by
 `i ↦ n-(i+1)`. -/
+@[simps! apply symm_apply]
 def revPerm : Equiv.Perm (Fin n) :=
-  (Involutive.toPerm fun i => ⟨n - (i + 1), tsub_lt_self i.pos (Nat.succ_pos _)⟩) fun i =>
-    ext <| by
-      dsimp only
-      rw [← tsub_tsub, tsub_tsub_cancel_of_le (Nat.add_one_le_iff.2 i.is_lt),
-        add_tsub_cancel_right]
+  Involutive.toPerm rev rev_involutive
 #align fin.rev Fin.revPerm
 
--- porting note: dot notation of the form `i.revPerm` is broken here and throughout
 @[simp]
-theorem val_revPerm (i : Fin n) : (revPerm i : ℕ) = n - (i + 1) :=
+theorem val_rev (i : Fin n) : (rev i : ℕ) = n - (i + 1) :=
   rfl
-#align fin.coe_rev Fin.val_revPerm
-
-theorem revPerm_involutive : Involutive (@revPerm n) :=
-  Involutive.toPerm_involutive _
-#align fin.rev_involutive Fin.revPerm_involutive
+#align fin.coe_rev Fin.val_rev
 
-theorem revPerm_injective : Injective (@revPerm n) :=
-  revPerm_involutive.injective
-#align fin.rev_injective Fin.revPerm_injective
+theorem rev_injective : Injective (@rev n) :=
+  rev_involutive.injective
+#align fin.rev_injective Fin.rev_injective
```

Comment:
These aligns should all use an `x` subscript, as they no longer align perfectly with the lean3 version.

Reply:
(or we should label this after-port, since it's a refactor)